pkg/discovery/globalnet/globalnet.go

```diff
@@ -227,3 +246,143 @@ func nextPowerOf2(n uint32) uint {
 	n++
 	return uint(n)
 }
+
+func CheckOverlappingCidr(networks map[string]*GlobalNetwork, resource overlapCheck, netconfig Netconfig) error {
+	var cidrlist []string
+	var cidr string
+	for k, v := range networks {
+		if resource == 0 {
+			cidrlist = v.ClusterCIDRs
+			cidr = netconfig.ClusterCIDR
+		} else if resource == 1 {
+			cidrlist = v.ServiceCIDRs
+			cidr = netconfig.ServiceCIDR
+		} else if resource == 2 {
+			cidrlist = v.GlobalCIDRs
+			cidr = netconfig.GlobalnetCIDR
+		}
```

Comment:
Use the type constants and a switch:

```
var cidrlist []string
var cidr string
switch overlappingFor {
case Cluster:
    cidrlist = v.ClusterCIDRs
    cidr = netconfig.ClusterCIDR
case Service:
    cidrlist = v.ServiceCIDRs
    cidr = netconfig.ServiceCIDR
case Global:
    cidrlist = v.GlobalCIDRs
    cidr = netconfig.GlobalnetCIDR
}

```

Reply:
Its just about personal code style with such a small loop. I have anyways changed it :)